### PR TITLE
support ssh

### PR DESCRIPTION
**Pull request by yungez**
support run playbook via ssh remotely

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -51,22 +53,34 @@ This extension helps to use [Ansible](https://www.ansible.com/) efficiently in V
...
+    1. Configure your remote server in `$HOME/.ssh/server.json` like below. Or follow wizard to fill in server information.
+       ```
+       [
+            {
+                "host": "your host",
+                "port": 22,
+                "user": "your user name",
+                "password": "your ssh password",
+                "key": "your private key"
+            }
```
[the remark this answers is not in the document]
Comment (yungez) [reply README.md]: user still need specify private key file

File: package.json [hunk 2]
```diff
@@ -47,6 +48,11 @@
         "command": "vscode-ansible.playbook-in-localansible",
         "title": "Run Ansible Playbook in Local Ansible",
         "category": "ansible"
+      },
+      {
+        "command": "vscode-ansible.ssh",
+        "title": "Run Ansible Playbook Remotely via SSH",
+        "category": "ansible"
```
[the remark this answers is not in the document]
Comment (yungez) [reply package.json]: sure

File: src/SSHConsole.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+		if (isWindows) {
+			modulePath = modulePath.replace(/\\/g, '\\\\');
+		}
+		const shellArgs = [
+			process.argv0,
+			'-e',
+			`require('${modulePath}').main()`,
+		];
+
+		if (isWindows) {
```
[the remark this answers is not in the document]
Comment (yungez) [reply src/SSHConsole.ts]: user don't need to install ssh.

File: src/SSHConsole.ts [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+		var envs = {
+			SSH_HOST: server.host,
+			SSH_PORT: String(server.port),
+			SSH_USER: server.user,
+			NODE_TLS_REJECT_UNAUTHORIZED: "0",
+			SSH_PASSWORD: server.password,
+			SSH_KEY: server.key
+		};
+
+		const terminal = vscode.window.createTerminal({
```
[the remark this answers is not in the document]
Comment (yungez) [reply src/SSHConsole.ts]: new terminal. it's consistent with all other terminal.

File: src/sshRunner.ts [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+import * as ssh from 'ssh2';
+import { TerminalBaseRunner } from './terminalBaseRunner';
+import * as vscode from 'vscode';
+import { SSHServer } from './interfaces';
+import * as utilities from './utilities';
+import { Constants } from './constants';
+import { TelemetryClient } from './telemetryClient';
+import * as path from 'path';
+import * as fs from 'fs-extra';
+import { openSSHConcole } from './SSHConsole';
```
[the remark this answers is not in the document]
Comment (yungez) [reply src/sshRunner.ts]: nice catch

File: src/sshRunner.ts [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+            // copy playbook
+            let destPlaybookFolder = this.getTargetFolder();
+
+            utilities.copyFileRemote(playbook, destPlaybookFolder, server, (err) => {
+                if (err) {
+                    return;
+                } else {
+                    // run playbook
+                    openSSHConcole(this._outputChannel, server).then((terminal) => {
+                        if (terminal) {
```
[the remark this answers is not in the document]
Comment (yungez) [reply src/sshRunner.ts]: no

File: src/sshRunner.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (zikalino) [reply src/sshRunner.ts]: i was trying to do search for "Console" ;-)

**Pull request merged**